tests/integration/node_compat_tests.rs

```diff
@@ -21,3 +21,11 @@ itest!(node_test_module_no_sanitizers {
   // exit_code: 123,
   http_server: true,
 });
+
+itest!(
+  node_process_beforeexit_exit_events_emitted_without_listeners {
+    args: "run node/process_beforeexit_exit_events.ts",
+    output: "node/process_beforeexit_exit_events.out",
+    exit_code: 0,
+  }
+);
```

Comment:
Great, can you add another test that uses both Web APIs and Node events and assert the correct event listeners order?